pkg/ingester/errors_test.go

```diff
@@ -709,3 +709,43 @@ func checkErrorWithStatusDetails(t *testing.T, details []any, expectedDetails *m
 		require.Equal(t, expectedDetails, errDetails)
 	}
 }
+
```

Comment:
nit: Feels a bit odd to put methods from a non-test type into the `_test.go` file. Can you add a comment explaining that this is only needed for tests?

Reply:
I have reverted this change, and put the methods back to `errors.go` with a comment stating that these methods are needed for tests only.